src/argilla/training/openai.py

```diff
@@ -0,0 +1,223 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging
+from typing import List, Union
+
+from argilla.datasets import TextClassificationSettings
+from argilla.training.base import ArgillaTrainerSkeleton
+from argilla.training.utils import filter_allowed_args
+from argilla.utils.dependency import require_version
+
+
+class ArgillaOpenAITrainer(ArgillaTrainerSkeleton):
+    _logger = logging.getLogger("ArgillaOpenAITrainer")
+    _logger.setLevel(logging.INFO)
+    _separator = "\n\n###\n\n"
+
+    require_version("openai")
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        import os
+
+        self.sleep_timer = 10
+        self.device = None
+        self.finetune_id = None
+
+        if self._seed is not None:
+            self._logger.warning("Seed is not supported for OpenAI. Ignoring seed for training.")
+
+        if self._model is None:
+            self._model = "curie"
```

Comment:
Maybe we can default `ada`? Just pointing it out because it's the fastest and cheaper OpenAI model to fine-tune. See https://platform.openai.com/docs/models/gpt-3